Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -270,11 +291,21 @@ public actor NetworkProtectionDeviceManager: NetworkProtectionDeviceManagement {
             throw NetworkProtectionError.couldNotGetInterfaceAddressRange
         }
 
+        let dns: [DNSServer]
+        switch dnsSettings {
+        case .default:
+            dns = [DNSServer(address: server.serverInfo.internalIP)]
+        case .custom(let servers):
+            dns = servers
+                .compactMap { IPv4Address($0) }
+                .map { DNSServer(address: $0) }
```

Comment:
Since we have two values for storing DNS settings, one for indicating whether to use custom DNS and another with the DNS servers themselves, it's possible for this to break if the DNS server array is unavailable. At that point the app will reach this code and end up with an empty DNS server array. It seems very unlikely, but on macOS users can modify user default values manually via the Terminal so anything's possible.

Should we fall back to using our own DNS server if this happens?